src/picongpu/include/particles/bremsstrahlung/Bremsstrahlung.tpp

```diff
@@ -0,0 +1,252 @@
+    const float_X deflectionProb = ionDensity * c * DELTA_T * scaledDeflectionDCS;
+
+    if(this->randomGen() < deflectionProb)
+    {
+        mom = this->scatterByTheta(mom, theta);
+        mom_norm = mom / momAbs;
+    }
+
+    /* non-radiative Bremsstrahlung */
+    const float_X kappaCutoff = math::min(this->softPhotonsCutoff / Ekin, float_X(1.0));
+    const float_X stoppingPower = ionDensity * c * this->stoppingPowerFunctor(Ekin, kappaCutoff);
+    const float_X newEkin = math::max(Ekin - stoppingPower * DELTA_T, float_X(0.0));
+    const float_X newEkin_norm = newEkin / (mass * c*c);
+    const float_X newMomAbs = mass * c * math::sqrt(newEkin_norm*newEkin_norm + float_X(2.0) * newEkin_norm);
+    const float_X deltaMom = newMomAbs - momAbs;
+    particle[momentum_] = (mom + deltaMom * mom_norm) * weighting;
+
+    /* photon emission */
+    const float_X delta = this->randomGen();
+    const float_X kappa = math::pow(kappaCutoff, float_X(1.0) - delta);
+    const float_X scalingFactor = -math::log(kappaCutoff);
+    const float_X emissionProb = scalingFactor * ionDensity * c * DELTA_T * this->scaledSpectrumFunctor(Ekin, kappa);
+
+    // raise a warning if the emission probability is too high.
+    if(picLog::log_level & picLog::CRITICAL::lvl)
+    {
+        if(emissionProb > float_X(photon::SINGLE_EMISSION_PROB_LIMIT))
+        {
+            const float_X Ekin_SI = Ekin * UNIT_ENERGY;
+            printf("[Bremsstrahlung] warning: emission probability is too high: \
+                    p = %g, at Ekin = %g keV, kappa = %g, ion density = %g m^-3\n",
+                emissionProb,
+                Ekin_SI * UNITCONV_Joule_to_keV,
+                kappa,
+                ionDensity / (UNIT_LENGTH*UNIT_LENGTH*UNIT_LENGTH));
+        }
+    }
+
+    if(this->randomGen() < emissionProb)
+    {
+        const float_X photonEnergy = kappa * Ekin;
+        this->photonMom = mom_norm * weighting * photonEnergy / c;
+        return 1;
+    }
+
+    return 0;
+}
+
+
+template<typename T_IonSpecies, typename T_ElectronSpecies, typename T_PhotonSpecies>
+template<typename Electron, typename Photon>
+DINLINE
+void Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::operator()
+    (Electron& electron, Photon& photon)
+{
+    namespace pam = PMacc::algorithms::math;
+
+    PMACC_AUTO(destPhoton,
+        PMacc::particles::operations::deselect<
+            boost::mpl::vector<
+                multiMask,
+                momentum
+            >
+        >(photon)
+    );
+    PMacc::particles::operations::assign( photon, electron );
+
+    const float3_X elMom = electron[momentum_];
+    const float_X weighting = electron[weighting_];
+    electron[momentum_] = elMom - this->photonMom; // ultra relativistic limit in terms of energy
+
+    /* photon emission angle */
+    const float_X mass = frame::getMass<FrameType>();
+    const float_X gamma = Gamma<>()(elMom / weighting, mass);
```

Comment:
`Gamma` is a struct that can be given to the implementation via template parameter as done for example in the pushers.